tests/coverage/extensions.py

```diff
@@ -0,0 +1,284 @@
+# SPDX-License-Identifier: Apache-2.0
+import os
+import yaml
+from ruamel.yaml import YAML
+
+enable_debug = False
+
+
+def error(msg):
+    print(f"ERROR: {msg}")
+
+
+def debug(msg):
+    if enable_debug:
+        print(f"DEBUG: {msg}")
+
+
+type_to_short_type = {
+    "required enumeration": "req",
+    "i8": "i8",
+    "i16": "i16",
+    "i32": "i32",
+    "i64": "i64",
```

Comment:
is there a way we can push these patterns into the parser for use across tools? I think we need them to go from in grammar literal to types. Let's just do that with the grammar.